packages/odyssey-react-mui/src/DataTable/useScrollIndication.tsx

```diff
@@ -0,0 +1,114 @@
+  SetStateAction,
+  useCallback,
+  useEffect,
+  useRef,
+} from "react";
+
+type UseScrollIndicationProps = {
+  tableOuterContainer: HTMLDivElement | null;
+  tableInnerContainer: HTMLDivElement | null;
+  setIsTableContainerScrolledToStart: Dispatch<SetStateAction<boolean>>;
+  setIsTableContainerScrolledToEnd: Dispatch<SetStateAction<boolean>>;
+  setTableInnerContainerWidth: Dispatch<SetStateAction<number | string>>;
+};
+export const useScrollIndication = ({
+  tableOuterContainer,
+  tableInnerContainer,
+  setIsTableContainerScrolledToStart,
+  setIsTableContainerScrolledToEnd,
+  setTableInnerContainerWidth,
+}: UseScrollIndicationProps) => {
+  const animationFrameIdRef = useRef<number | null>(null);
+
+  const resizeObserverRef = useRef<ResizeObserver | null>(null);
+
+  const checkScrollIndicators = useCallback(() => {
+    if (!tableOuterContainer || !tableInnerContainer) return;
+
+    const containerWidth = tableOuterContainer.clientWidth;
+    const contentWidth = tableInnerContainer.scrollWidth;
+    const containerStartScrollPosition = tableInnerContainer.scrollLeft;
+    const containerEndScrollPosition =
+      containerStartScrollPosition + containerWidth;
+
+    setIsTableContainerScrolledToStart(containerStartScrollPosition <= 16);
+    setIsTableContainerScrolledToEnd(
+      containerEndScrollPosition >= contentWidth - 16,
+    );
+  }, [
+    tableInnerContainer,
+    tableOuterContainer,
+    setIsTableContainerScrolledToEnd,
+    setIsTableContainerScrolledToStart,
+  ]);
+
+  useEffect(() => {
+    if (resizeObserverRef.current) return; // Avoid creating multiple observers
+
+    let debounceTimer: ReturnType<typeof setTimeout>;
+
+    resizeObserverRef.current = new ResizeObserver(() => {
+      clearTimeout(debounceTimer);
+      debounceTimer = setTimeout(() => {
+        if (!animationFrameIdRef.current) {
+          animationFrameIdRef.current = requestAnimationFrame(() => {
+            checkScrollIndicators();
+            animationFrameIdRef.current = null;
+          });
+        }
+
+        setTableInnerContainerWidth(tableInnerContainer?.clientWidth ?? "100%");
+      }, 100); // debounce delay
+    });
+
+    if (tableOuterContainer) {
+      resizeObserverRef.current.observe(tableOuterContainer);
+    }
+
+    return () => {
+      if (resizeObserverRef.current) {
+        resizeObserverRef.current.disconnect();
+        resizeObserverRef.current = null;
+      }
+      clearTimeout(debounceTimer); // Ensure timer is cleared on component unmount
+      if (animationFrameIdRef.current) {
+        cancelAnimationFrame(animationFrameIdRef.current);
+        animationFrameIdRef.current = null;
+      }
+    };
+  }, [
+    checkScrollIndicators,
+    setTableInnerContainerWidth,
+    tableOuterContainer,
+    tableInnerContainer,
+    setIsTableContainerScrolledToStart,
+    setIsTableContainerScrolledToEnd,
+  ]);
+
+  useEffect(() => {
+    const handleScroll = () => checkScrollIndicators();
```

Comment:
Do you need this variable? I believe you can just pass `checkScrollIndicators` as arg to calls below